types/src/v0/impls/stake_table.rs

```diff
@@ -222,6 +224,26 @@ impl StakeTableState {
         }
     }
 
+    pub fn hash(&self) -> Result<StakeTableStateHash, bincode::Error> {
+        let mut hasher = alloy::primitives::Keccak256::new();
+        for (address, validator) in &self.validators {
+            hasher.update(address);
+            hasher.update(bincode::serialize(&validator)?);
+        }
+        for key in self.used_bls_keys.iter().sorted() {
+            hasher.update(bincode::serialize(&key)?);
```

Comment:
@sveitser @alxiong will there be any encoding problem here if we are hashing it in the contract?

Reply:
The flow would be to get the stake table state from another query node, hash it, then check the has matches what's stored in the LC contract.